src/main/scala/com/scality/clueso/query/MetadataQueryExecutor.scala

```diff
@@ -0,0 +1,255 @@
+package com.scality.clueso.query
+
+import java.util.concurrent.Executors
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.{concurrent => juc}
+
+import com.scality.clueso._
+import com.scality.clueso.query.cache.SessionCacheManager
+import com.typesafe.scalalogging.LazyLogging
+import org.apache.spark.SparkContext
+import org.apache.spark.clueso.metrics.SearchMetricsSource
+import org.apache.spark.sql.expressions.Window
+import org.apache.spark.sql.functions.{col, _}
+import org.apache.spark.sql.{DataFrame, Dataset, Row, SparkSession}
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.{Await, ExecutionContext, Future, TimeoutException}
+
+class MetadataQueryExecutor(spark : SparkSession, config : CluesoConfig) extends LazyLogging {
+
+  import MetadataQueryExecutor._
+
+  implicit val _config = config
+  implicit val ec = ExecutionContext.fromExecutor(Executors.newFixedThreadPool(10))
+
+  var lockedBucketName : Option[String] = None
+
+  SearchMetricsSource(spark, config)
+
+  val metricsRegisterCancel = new AtomicBoolean(false)
+
+  val metricsRegisterThread = Future {
+    while (!metricsRegisterCancel.get()) {
```

Comment:
This thread runs on the driver program and it has been only counting driver program metrics and search counts. 
It is very tricky to define dynamic metrics upfront – ideally we would have a search measurement per bucket, but this doesn't seem to be very possible because the MetricsSystem is created before buckets are known and adding new metrics doesn't seem to have effect in Graphite reporter.

This code is currently commented out as we have some problems that seemed to be related with metrics reporting – I will enable this one we have all stable and see if there's a way to keep track of basic metrics.